src/Packetery/Module/Labels/LabelPrintPacketData.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Packetery\Module\Labels;
+
+use LogicException;
+use Packetery\Core\Entity\Order;
+
+class LabelPrintPacketData {
+
+	/**
+	 * @var LabelPrintPacketDataItem[]
+	 */
+	protected array $items;
+
+	/**
+	 * @var array<int, Order>|null
+	 */
+	private ?array $ordersByOrderIdCache = null;
+
+	public function __construct() {
+		$this->items = [];
+	}
+
+	public function addItem( Order $order, string $packetId ): void {
+		$this->items[]              = new LabelPrintPacketDataItem( $order, $packetId );
+		$this->ordersByOrderIdCache = null;
+	}
+
+	/**
+	 * @return LabelPrintPacketDataItem[]
+	 */
+	public function getItems(): array {
+		return $this->items;
+	}
+
+	/**
+	 * @return string[]
+	 */
+	public function getPacketIds(): array {
+		$packetIds = [];
+		foreach ( $this->items as $item ) {
+			$packetIds[] = $item->getPacketId();
+		}
+
+		return $packetIds;
+	}
+
+	public function isEmpty(): bool {
+		return $this->items === [];
+	}
+
+	public function getCount(): int {
+		return count( $this->items );
+	}
+
+	public function getFirstItem(): LabelPrintPacketDataItem {
+		if ( $this->isEmpty() ) {
+			throw new LogicException( 'Cannot get first item from empty collection.' );
+		}
+
+		return $this->items[0];
+	}
+
+	/**
+	 * @return array<int, Order>
+	 */
+	private function getOrdersByOrderId(): array {
+		if ( $this->ordersByOrderIdCache === null ) {
```

Comment:
použil bych statickou cache v rámci té metody -  děláme to tak už na jiných místech, např v \Packetery\Module\Order\GridExtender::getOrderByIdCached.  Nemusíš pak zavádět novou privátní property $ordersByOrderIdCache v rámci celé třídy.

Reply:
privátní metody přece unit testy neprokrýváme.  Jaké testy tedy myslíš ?
Tak jak je metoda navržena a k čemu je určena nějak nevidím důvod, proč bychom měli potřebovat tu cache průběžně promazávat.  Nejdřív získáme kolekci, kterou předáme \Packetery\Module\Order\LabelPrint::requestCarrierLabels kde z ní data vytahujeme.